x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorFilteringAction.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.action;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.application.connector.Connector;
+import org.elasticsearch.xpack.application.connector.ConnectorFiltering;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+public class UpdateConnectorFilteringAction extends ActionType<UpdateConnectorFilteringAction.Response> {
+
+    public static final UpdateConnectorFilteringAction INSTANCE = new UpdateConnectorFilteringAction();
+    public static final String NAME = "cluster:admin/xpack/connector/update_filtering";
+
+    public UpdateConnectorFilteringAction() {
+        super(NAME, UpdateConnectorFilteringAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest implements ToXContentObject {
+
+        private final String connectorId;
+        private final List<ConnectorFiltering> filtering;
+
+        public Request(String connectorId, List<ConnectorFiltering> filtering) {
+            this.connectorId = connectorId;
+            this.filtering = filtering;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.connectorId = in.readString();
+            this.filtering = in.readOptionalCollectionAsList(ConnectorFiltering::new);
+        }
+
+        public String getConnectorId() {
+            return connectorId;
+        }
+
+        public List<ConnectorFiltering> getFiltering() {
+            return filtering;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
```

Comment:
Should we validate that `connectorId` is non null/empty here?

Reply:
Added :)